giza_datasets/cache_manager.py

```diff
@@ -0,0 +1,94 @@
+import polars as pl
+import os
+import hashlib
+
+class CacheManager:
+    def __init__(self, cache_dir):
+        """
+        Initializes the CacheManager with a specified cache directory.
+        
+        Args:
+            cache_dir (str): The directory path where cached datasets will be stored.
+        """
+        self.cache_dir = cache_dir
+        if not os.path.exists(cache_dir):
+            os.makedirs(cache_dir)
+
+    def _generate_hash(self, parameters):
+        """
+        Generates a hash based on the given parameters usefull to avoid collisions in the cache.
+        
+        Args:
+            parameters (dict): Parameters used to generate the hash.
+            
+        Returns:
+            str: A hash string derived from the parameters.
+        """
+        param_str = str(parameters).encode('utf-8')
+        return hashlib.sha256(param_str).hexdigest()
```

Comment:
When you mention collision, basically we are creating a hash from a name (or thats what I get following the code). This hash is deterministic and thats how we can download it later.

* How are this collision produced if we do not use the hash?
* Wouldn't the name suffice for this?

```python
>>> a = str("dataset").encode("utf-8")
>>> hashlib.sha256(a).hexdigest()
'b277fd623676a525c29b9eb155afc8c9010681814ceafb2d7627f47b9a232576'
>>> hashlib.sha256(a).hexdigest()
'b277fd623676a525c29b9eb155afc8c9010681814ceafb2d7627f47b9a232576'
```

If the name is unique does it make a difference to hash it?

Reply:
I have removed all hashing functionality from the cache